deno-ts/src/physical_printer.ts

```diff
@@ -0,0 +1,97 @@
+import { exists } from "jsr:@std/fs/exists";
+import {
+  DiagramMetadata,
+  FlowchartDiagram,
+  IDiagramMetadata,
+} from "./documents.ts";
+import {
+  PhysicalPrinter,
+  PrinterDriverFactory,
+  PrintMetadata,
+  PrintQueue,
+  Toner,
+} from "./printing.ts";
+import { Mutex } from "npm:async-mutex";
+
+class DiagramPhysicalPrinter {
+  private _physicalPrinter: PhysicalPrinter;
+  private _printQueue: PrintQueue;
+
+  constructor(physicalPrinter?: PhysicalPrinter, printQueue?: PrintQueue) {
+    this._physicalPrinter = physicalPrinter || new PhysicalPrinter();
+    this._printQueue = printQueue || new PrintQueue(this._physicalPrinter);
+  }
+
+  async doPrint(
+    diagram: FlowchartDiagram,
+    info: DiagramMetadata,
+    targetFilename: string,
+  ): Promise<boolean> {
+    const printerDriver = PrinterDriverFactory.getInstance()
+      .createDriverForPrint();
+    printerDriver.setDiagram(diagram);
+
+    const data = new PrintMetadata(info.fileType);
+    const mutex = new Mutex();
+    let success = false;
+
+    try {
+      await mutex.acquire();
+
+      const hasEnoughToner =
```

Comment:
This variable 'hasEnoughToner' makes sense but I don't think I've got it in the other language translations. I'd prefer to inline it if possible.